primitives/src/assets/currencies.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023-2024 Forecasting Technologies LTD.
+//
+// This file is part of Zeitgeist.
+//
+// Zeitgeist is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at
+// your option) any later version.
+//
+// Zeitgeist is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+
+/// The `CurrencyClass` enum represents all non-ztg CurrencyClass
+// used in orml-tokens
+#[cfg_attr(feature = "std", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
+#[derive(Clone, Copy, Debug, Decode, Eq, Encode, MaxEncodedLen, PartialEq, TypeInfo)]
+pub enum CurrencyClass<MI> {
+    // All "Old" variants will be removed once the lazy migration from
+    // orml-tokens to pallet-assets is complete
+    #[codec(index = 0)]
+    OldCategoricalOutcome(MI, CategoryIndex),
+
+    #[codec(index = 2)]
+    OldCombinatorialOutcome,
+
+    #[codec(index = 1)]
+    OldScalarOutcome(MI, ScalarPosition),
+
+    #[codec(index = 6)]
+    OldParimutuelShare(MI, CategoryIndex),
+
+    #[codec(index = 3)]
+    OldPoolShare(PoolId),
+
+    // Type can not be compacted as it is already used uncompacted in the storage
+    #[codec(index = 5)]
+    ForeignAsset(u32),
```

Comment:
I just used lexicographical ordering for esthetic reasons.

Reply:
p, q, r, s, t, u...

`OldScalarOutcome`
`OldParimutuelShare`

Reply:
I personall would not order this lexicographical, because we most likely will add new enum fields at the end regardless of the alphabet, because it avoids a storage migration. However, even if the order of `codec(index = ...)` is the common counting order, we might want to keep it in, because it allows the error to be tracked better, if it is shown as a number on the front end.

Reply:
Obviously I would not maintain lexicographical ordering after it's deployed. Those are new types yet unused. Also, some of the indices will still be missing as not all variants from the `Asset` enum are supported. Whenever someone has to track the error, the can explicitly see in `the codec(index)` and know which variant caused the error.

Do prefer this variant? https://github.com/zeitgeistpm/zeitgeist/pull/1181/commits/53fec66f630ea7d21ea1a2832a5463777182a5e2

Reply:
> Do prefer this variant? https://github.com/zeitgeistpm/zeitgeist/commit/53fec66f630ea7d21ea1a2832a5463777182a5e2

Much better <3